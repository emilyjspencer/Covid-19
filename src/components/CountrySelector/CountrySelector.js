import React, { useState, useEffect } from 'react';
import { NativeSelect, FormControl } from '@material-ui/core';
import { fetchCountries } from '../../api';

import styles from './CountrySelector.module.css';



    const CountrySelector = ({ handleCountryChange }) => {
    const [countries, setCountries] = useState([]);

    useEffect(() => {
        const fetchApi = async () => {
            setCountries(await fetchCountries());
        }

            fetchApi();
        }, []);

        
    

    return(
        <FormControl className={styles.formControl}>
            <NativeSelect id="dropdown2" defaultValue="" onChange={(e) => handleCountryChange(e.target.value)} >
                <option id="dropdown" color="white" value="">Global Daily Data - click the dropdown to select a country</option>
                {countries.map((country, index) => <option key={index} value={country}>{country}</option>)}
                <h1>Test</h1>
            </NativeSelect>
        </FormControl>
    )
}


export default CountrySelector;
