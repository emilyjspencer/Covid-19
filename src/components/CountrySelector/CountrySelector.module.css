body {
    background-color: rgb(250, 250, 250);
}

.container {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    background-color: rgb(83, 211, 200);
}

#dropdown {
    background-color: white;
    color: white;
}

#dropdown2 {
    background-color: white;
    color: white;
}

.MuiNativeSelect-root {
    background-color: white;
    color: white
}

.MuiNativeSelect-select { 
    background-color: white;
    color: white;
}