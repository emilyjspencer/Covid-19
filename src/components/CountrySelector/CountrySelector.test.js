import React from 'react';
import { shallow } from 'enzyme';
import CountrySelector from './CountrySelector';

describe('CountrySelector', () => {

    let wrapper;

    beforeEach(() => wrapper = shallow(<CountrySelector />));

    it('the dropdown is by default global', () => {
      expect(wrapper.find('#dropdown').text()).toEqual('Global Daily Data - click the dropdown to select a country');
    });
});