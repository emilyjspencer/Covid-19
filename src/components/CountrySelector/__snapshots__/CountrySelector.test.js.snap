// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountrySelector renders correctly 1`] = `ShallowWrapper {}`;
